src/qtt/instrument_drivers/virtualAwg/awgs/Tektronix5014C.py

```diff
@@ -124,18 +250,26 @@ def _set_sequence(self, channels, sequence):
                     self.__awg.set_sqel_waveform("", channel, row_index + 1)
         self.__awg.set_sqel_goto_state(request_rows, 1)
 
-    def _upload_waveforms(self, names, waveforms, file_name='default.awg'):
+    def _upload_waveforms(self, names: List[str], waveforms: List[np.ndarray],
+                          file_name: Optional[str] = 'default.awg') -> None:
+        """ Upload the waveforms to the AWG. Creates an AWG file and then uploads it to the AWG.
+
+        Args:
+            names: A list with the waveform names.
+            waveforms: A list containing the waveform data.
+            file_name: The name of the AWG file.
+        """
         pack_count = len(names)
         packed_waveforms = dict()
         [wfs, m1s, m2s] = list(map(list, zip(*waveforms)))
         for i in range(pack_count):
             name = names[i]
             package = self.__awg.pack_waveform(wfs[i], m1s[i], m2s[i])
             packed_waveforms[name] = package
-        offset = self.__settings[5].get()
-        amplitude = self.__settings[4].get()
-        marker_low = self.__settings[2].get()
-        marker_high = self.__settings[3].get()
+        offset = self.__settings[4].get()
```

Comment:
These indexes are not really safe, when someone changes the order in the constructor...
A descriptive name would be more.... descriptive

Reply:
Yes, I agree and changed it. I also removed the marker_delay setting because it is not used.